concordium-std/src/state_btree.rs

```diff
@@ -0,0 +1,2111 @@
+        let mut tree = state_builder.new_btree_set_degree::<2, _>();
+        for n in 0..4 {
+            tree.insert(n);
+        }
+        tree.remove(&3);
+        claim!(tree.remove(&1));
+
+        claim!(tree.contains(&0));
+        claim!(!tree.contains(&1));
+        claim!(tree.contains(&2));
+    }
+
+    /// Removing from a the root node which is at minimum keys, taking a child
+    /// from its higher child.
+    ///
+    /// - Builds a tree of the form: [[0], 1, [2, 3]]
+    /// - Remove 1
+    /// - Expecting a tree of the form: [[0], 2, [3]]
+    #[concordium_test]
+    fn test_btree_remove_from_root_in_three_node_taking_key_from_higher_child() {
+        let mut state_builder = StateBuilder::open(StateApi::open());
+        let mut tree = state_builder.new_btree_set_degree::<2, _>();
+        for n in 0..4 {
+            tree.insert(n);
+        }
+        claim!(tree.contains(&1));
+        claim!(tree.remove(&1));
+        claim!(!tree.contains(&1));
+    }
+
+    /// Removing from a the root node which is at minimum keys, taking a child
+    /// from its lower child.
+    ///
+    /// - Builds a tree of the form: [[0, 1], 2, [3]]
+    /// - Remove 1
+    /// - Expecting a tree of the form: [[0], 1, [3]]
+    #[concordium_test]
+    fn test_btree_remove_from_root_in_three_node_taking_key_from_lower_child() {
+        let mut state_builder = StateBuilder::open(StateApi::open());
+        let mut tree = state_builder.new_btree_set_degree::<2, _>();
+        for n in (0..4).into_iter().rev() {
+            tree.insert(n);
+        }
+        claim!(tree.contains(&2));
+        claim!(tree.remove(&2));
+        claim!(!tree.contains(&2));
+    }
+
+    /// Test iteration of the set.
+    #[concordium_test]
+    fn test_btree_iter() {
+        let mut state_builder = StateBuilder::open(StateApi::open());
+        let mut tree = state_builder.new_btree_set_degree::<2, _>();
+        let keys: Vec<u32> = (0..15).into_iter().collect();
+        for &k in &keys {
+            tree.insert(k);
+        }
+        let iter_keys: Vec<u32> = tree.iter().map(|k| k.clone()).collect();
+        claim_eq!(keys, iter_keys);
+    }
+
+    /// Testcase for dublicate keys in the set bug. Due to an edgecase where the
+    /// key moved up as part of splitting a child node, is equal to the
+    /// inserted key.
+    ///
+    /// - Builds a tree of the form: [[0, 1, 2], 3, [4]]
+    /// - Insert 1 (again) causing the [0,1,2] to split, moving 1 up to the
+    ///   root.
+    /// - Expecting a tree of the form: [[0], 1, [2], 3, [4]]
+    #[concordium_test]
+    fn test_btree_insert_present_key() {
+        let mut state_builder = StateBuilder::open(StateApi::open());
+        let mut tree = state_builder.new_btree_set_degree::<2, _>();
+        for n in [0, 3, 4, 1, 2].into_iter() {
+            tree.insert(n);
+        }
+        claim!(!tree.insert(1));
+    }
+
+    #[allow(deprecated)]
```

Comment:
Why do we need this?

Reply:
We need this, because `#[concordium_quickcheck]` is located in a deprecated module